ompi/mca/common/ompio/common_ompio_aggregators.c

```diff
@@ -1303,12 +1303,14 @@ int mca_common_ompio_prepare_to_group(ompio_file_t *fh,
                                            fh->f_comm);
     if ( OMPI_SUCCESS != ret ) {
         opal_output (1, "mca_common_ompio_prepare_to_group: error in ompi_fcoll_base_coll_allgather_array\n");
+        free(start_offsets_lens_tmp);
         goto exit;
     }
     end_offsets_tmp = (OMPI_MPI_OFFSET_TYPE* )malloc (fh->f_init_procs_per_group * sizeof(OMPI_MPI_OFFSET_TYPE));
     if (NULL == end_offsets_tmp) {
         opal_output (1, "OUT OF MEMORY\n");
-        goto exit;
+        free(start_offsets_lens_tmp);
+        return OMPI_ERR_OUT_OF_RESOURCE;
     }
```

Comment:
the same issue here. The arrays are actually freed in the layer above, even in case of an error. THis now would lead to a double free.

Reply:
I believe this is correct. The `free` and return happens before the out parameters are assigned so there is no way the upper layers may release that memory (the assignment happens on line 1323).

Reply:
ok, you are actually correct in this case.